.github/workflows/ci.yml

```diff
@@ -187,3 +187,28 @@ jobs:
 
     - name: Check formatting
       run: cargo fmt -- --check
+
+  rustsemver:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@a40b8845c0683271d9f53dfcb887a7e181d3918b # 0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - uses: actions/checkout@v3
+
+      - name: Install rust nightly
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+
+      - name: Install Semver Checks
+        run:  cargo install cargo-semver-checks
```

Comment:
Installing this on every CI run seems pretty wasteful. Can you add a cache please?

https://github.com/Swatinem/rust-cache